flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BoundedBlockingSubpartition.java

```diff
@@ -266,6 +266,13 @@ public int unsynchronizedGetNumberOfQueuedBuffers() {
         return 0;
     }
 
+    @Override
+    public int getNumberOfQueuedBuffers() {
+        synchronized (lock) {
+            return getBuffersInBacklogUnsafe();
```

Comment:
are you sure this is correct? It looks to me like `getBuffersInBacklogUnsafe()` counter is ever increasing one?

Besides, even if it had indeed thousands of buffers in backlog spilled on disks, I don't think we should be using that number for debloating... Maybe we should cap this returned number with `taskmanager.network.memory.max-buffers-per-channel`? And also cup the sum the number of total number of buffers in use/backlog with `#exclusiveBuffers * #channels + #floatingBuffers`?

Furthermore, debloating with `BoundedBlockingSubpartition` probably doesn't matter at the moment, as those partitions do not support checkpointing and are used only in batch.

Reply:
I actually understand that it doesn't make sense for `BoundedBlockingSubpartition` at the moment. You right that my implementation is wrong. I will just do the same behaviour which we have for `unsynchronizedGetNumberOfQueuedBuffers` namely just return 0